### PR TITLE
fix(typings): migrate 'auth' state slice to TypeScript

### PR DESCRIPTION
**Pull request by smashercosmo**
**Summary**

I took a stab on migrating 'auth' state slice to TypeScript. I did some things differently from how it's done for other state slices (explained in comments), but if you don't like it, I can rewrite it your way.

### CODE REVIEW

File: packages/netlify-cms-core/src/actions/auth.ts [hunk 1]
```diff
@@ -115,3 +118,11 @@ export function logoutUser() {
     });
   };
 }
+
+export type AuthAction = ReturnType<
```
Comment (smashercosmo) [inline packages/netlify-cms-core/src/actions/auth.ts:122]: Defining action type like this is much more type safe than
https://github.com/netlify/netlify-cms/blob/master/packages/netlify-cms-core/src/types/redux.ts#L402-L413

File: packages/netlify-cms-core/src/reducers/__tests__/auth.spec.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import { fromJS } from 'immutable';
+import { authenticating, authenticate, authError, logout } from '../../actions/auth';
+import auth, { defaultState } from '../auth';
+
+describe('auth', () => {
+  it('should handle an empty state', () => {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+    // @ts-ignore
```
Comment (smashercosmo) [inline packages/netlify-cms-core/src/reducers/__tests__/auth.spec.ts:8]: Reducer is now strongly typed, so it doesn't accept empty action

File: packages/netlify-cms-core/src/reducers/auth.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  AUTH_REQUEST,
+  AUTH_SUCCESS,
+  AUTH_FAILURE,
+  AUTH_REQUEST_DONE,
+  LOGOUT,
+  AuthAction,
+} from '../actions/auth';
+import { StaticallyTypedRecord } from '../types/immutable';
+
+export type Auth = StaticallyTypedRecord<{
```
Comment (smashercosmo) [inline packages/netlify-cms-core/src/reducers/auth.ts:13]: I think it's better to define state slice type right in reducer file and then import it to types/redux.ts. Currently all state slices types are defined in types/redux.ts, which makes redux.ts file very bloated.

**Comment by smashercosmo** [comment]
I have no idea why git can't properly diff renamed auth.spec.ts and reducers/auth.ts 🤔

**Review by erezrokah — APPROVED** [review]
Thanks @smashercosmo, looks great!

**Comment by erezrokah** [comment]
> I have no idea why git can't properly diff renamed auth.spec.ts and reducers/auth.ts

It does that when there are many changes. You can workaround it by doing an initial commit just for the rename, and another for the refactoring.

**Pull request merged**